src/main/java/com/extremelyd1/listener/ItemListener.java

```diff
@@ -53,4 +56,55 @@ public void onPlayerDropItem(PlayerDropItemEvent e) {
             e.setCancelled(true);
         }
     }
+
+    /**
+     * Caters for collecting milk, water or lava
+     *
+     * @param e - PlayerBucketFillEvent
+     */
+    @EventHandler
+    public void onBucketFill(PlayerBucketFillEvent e) {
+        if (!game.getState().equals(Game.State.IN_GAME)) {
+            e.setCancelled(true);
+            return;
+        }
+
+        game.onMaterialCollected(e.getPlayer(), e.getItemStack().getType());
+    }
+
+    /**
+     * This event caters for the collection of fishes into buckets, and the collection of mushroom stew from mooshrooms
```

Comment:
```suggestion
     * This event caters for the collection of fish into buckets, and the collection of mushroom stew from mooshrooms
```
The plural of fish is fish 😉

Reply:
removed java docs so this no longer applies, but every day is a school day :P